internal/vds/vds.cpp

```diff
@@ -405,6 +408,80 @@ struct vdsbuffer fetch_slice_metadata(
     return buffer;
 }
 
+struct vdsbuffer fetch_fence(
+    const std::string& vds,
+    const std::string& credentials,
+    const std::string& coordinate_system,
+    const float* horizon,
+    size_t npoints
+) {
+    OpenVDS::Error error;
+    OpenVDS::VDSHandle handle = OpenVDS::Open(vds, credentials, error);
+
+    if (error.code != 0) {
+        throw std::runtime_error("Could not open VDS: " + error.string);
+    }
+
+    auto accessManager = OpenVDS::GetAccessManager(handle);
+    auto const *layout = accessManager.GetVolumeDataLayout();
+    const auto dimension_map =
+            layout->GetVDSIJKGridDefinitionFromMetadata().dimensionMap;
+
+    unique_ptr< float[][OpenVDS::Dimensionality_Max] > horizn(
+        new float[npoints][OpenVDS::Dimensionality_Max]{0}
+    );
+
+    auto coordinate_transformer = OpenVDS::IJKCoordinateTransformer(layout);
+    auto transform_coordinate = [&] (const float x, const float y) {
+        if (coordinate_system == "ij") {
+            return OpenVDS::Vector<double, 3> {x, y, 0};
+        }
+        else if (coordinate_system == "ilxl") {
+            return coordinate_transformer.AnnotationToIJKPosition({x, y, 0});
+        }
+        else if (coordinate_system == "cdp") {
+            return coordinate_transformer.WorldToIJKPosition({x, y, 0});
+        }
+        else {
+            const auto msg =
+                    "Coordinate system not recognized: " + coordinate_system;
+            throw std::runtime_error(msg);
+        }
+    };
+
+    for (size_t i = 0; i < npoints; i++) {
+        const float x = *(horizon++);
+        const float y = *(horizon++);
+
+        const auto coordinate = transform_coordinate(x, y);
+
+        horizn[i][dimension_map[0]] = coordinate[0];
+        horizn[i][dimension_map[1]] = coordinate[1];
+    }
+
+    // TODO: Verify that trace dimension is always 0
+    auto size = accessManager.GetVolumeTracesBufferSize(npoints, 0);
+
+    auto *data = new char[size];
+
+    auto request = accessManager.RequestVolumeTraces(
```

Comment:
Yes, this is very weird. Calling RequestVolumeTraces with a pre-allocated buffer is in fact an async call, but we do call waitForCompletion(), so all should be good.

Could be that the -race argument causes the tests to be run with a different configuration in order to trigger race conditions. And perhaps the race condition is not caught since it happens in the C++ code. 

Just making some guesses here, this will certainly need deeper investigation.

Reply:
I am not able to reproduce this issue